x/src/cargo.rs

```diff
@@ -0,0 +1,94 @@
+use crate::Result;
+use anyhow::anyhow;
+use std::{
+    ffi::{OsStr, OsString},
+    path::Path,
+    process::{Command, Output, Stdio},
+};
+
+pub struct Cargo {
+    inner: Command,
+    pass_through_args: Vec<OsString>,
+}
+
+impl Cargo {
+    pub fn new<S: AsRef<OsStr>>(command: S) -> Self {
+        let mut inner = Command::new("cargo");
+        inner.arg(command);
```

Comment:
Unfortunately that's not exactly the case. `arg` returns a `&mut Command`, not a `Command` so we'll end up with a type mismatch when trying to stick it into `Self { inner }` since the field type is a `Command` not a `&mut Command`.